### DISCUSSION TITLE
WebApp shows Error 500. It says This page isn't working at the moment [500 Error]

### DISCUSSION THREAD

**Discussion by yeggan** [microsoft/PubSec-Info-Assistant#814] [Evening (PST) Office Hours (Q&A)]
Hi 
I deployed the code in codespace without issue then I deployed in the Azure DevOps Pipeline for CI/CD, but I'm encountering some issues.

- Since I don’t have Enra Admin privileges, I created two distinct application registrations to ensure the application functions correctly within the DevOps environment: Infoasst Mgmt - DEV and Infoasst Web Access - DEV.
- To resolve the error: Deployment Name: "$(AZURE_OPENAI_CHATGPT_DEPLOYMENT)"): performing│ CreateOrUpdate: unexpected status 400 with error, I created an OpenAI instance named infoasst-aoai-host in the another resource group' via the Portal. Based on the available quota, I defined the GPT-4 model 0613 for text embedding mode. then I edited the parameter in scripts/environments/shared-ia.env to include the parameters mentioned above.
-I ran the Azure DevOps pipeline, which completed successfully and created all resources in Azure.

However, when I check the web interface, I encounter the following error: "This page isn't working at the moment. infoasst-web-abcde.azurewebsites.net can't currently handle this request.[500 Error]"

It’s important to note that after the resources were created, I edited the Web URI redirect to https://infoasst-web-abcde.azurewebsites.net/.auth/login/aad/callback based on random characters. 
When trying to access the web app, the logs show an Error 500: "Unable to download OpenID Connect Configuration."

Could you please help me resolve this issue?

**Comment by dayland** [comment]
Please see this documentation for how to manually set up the App Registrations and App Service Authentication settings... https://github.com/microsoft/PubSec-Info-Assistant/blob/main/docs/deployment/manual_app_registration.md and this Issue dialog for more details. https://github.com/microsoft/PubSec-Info-Assistant/discussions/457

**Reply by yeggan** [reply]
Thank you for your reply.

As I mentioned in my original request, I **successfully deployed** the repo in **Codespaces**. In Codespaces, I followed the App Registrations and App Service Authentication settings via the page you shared.

When I copied the repo to **Azure DevOps,** I registered two new _MGMTAPP_ and _WEBAPP_ applications and created a _CLIENT_ID_ for my Azure Pipeline environment.
In Codespaces, I defined the parameters inside the code in the **scripts/inf-create.sh** file (lines 63-68). However, in Azure DevOps, I defined these parameters **as variables** in the pipeline for security reasons, and the pipeline completed successfully.

When I checked the web interface, I encountered the following error: "**This page isn't working at the moment. infoasst-web-abcde.azurewebsites.net can't currently handle this request. [500 Error]**". 
The logs show an Error 500: "Unable to download **OpenID Connect Configuration."**

Below is a table listing the variables I used.


<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/yjafary/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<link rel=themeData
href="file:///C:/Users/yjafary/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/yjafary/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=EN-AU style='tab-interval:36.0pt;word-wrap:break-word'>


<ol start=1 type=1>
 <ul type=circle>
  <li class=MsoNormal style='color:#2F5597;mso-margin-top-alt:auto;mso-margin-bottom-alt:
      auto;mso-list:l0 level2 lfo1;tab-stops:list 72.0pt'><span
      style='mso-fareast-font-family:"Times New Roman";mso-ligatures:none;
      mso-fareast-language:EN-AU'>Edited Variables in the Pipeline and
      currently the values:<o:p></o:p></span></li>
 </ul>
</ol>


1 | TENANT_ID
-- | --
2 | SUBSCRIPTION_ID
3 | environment
4 | TF_BACKEND_RESOURCE_GROUP
5 | TF_BACKEND_STORAGE_ACCOUNT
6 | TF_BACKEND_ACCESS_KEY
7 | TF_BACKEND_CONTAINER
8 | CONTAINER_REGISTRY_ADDRESS
9 | CLIENT_ID
10 | CLIENT_SECRET
11 | SERVICE_PRINCIPAL_ID
12 | AZURE_OPENAI_SERVICE_NAME
13 | AZURE_OPENAI_SERVICE_KEY
14 | AZURE_OPENAI_CHATGPT_DEPLOYMENT
15 | AD_WEBAPP_CLIENT_ID
16 | AD_MGMTAPP_CLIENT_SECRET
17 | AD_MGMTAPP_CLIENT_ID
18 | AD_MGMT_SERVICE_PRINCIPAL_ID




</body>

</html>
